File: Film.java

public class Film
{
    private int duree;
    private String dateSortie;
    private Categorie categorie;

    public Film()
    {
        // initialisation des variables d'instance
        duree = 0;
        dateSortie = "2001/01/01";
    }

    public int getDuree()
    {
        return this.duree;
    }
    
    public void setDuree(int duree)
    {
        this.duree = duree;
    }
    
    public String getDateSortie()
    {
        return this.dateSortie;
    }
    
    public void setDateSortie(String dateSortie)
    {
        this.dateSortie = dateSortie;
    }
    
    public void setCategorie(Categorie categorie)
    {
        this.categorie = categorie;
    }
    
    public Categorie getCategorie()
    {
        return this.categorie;
    }
    
    public String description()
    {
        return "Je suis un film de catégorie " + this.categorie.getNom() + " et de durée " + this.duree + " sorti en " + this.dateSortie;
    }
    
}

File: film.py
from categorie import Categorie

class Film:
    def __init__(self):
        self._duree = 0
        self._date_sortie = "2001/01/01"
        self._categorie = None 
        self._realisateur = None

    def get_duree(self) -> int:
        return self._duree

    def set_duree(self, duree: int):
        self._duree = duree

    def get_date_sortie(self) -> str:
        return self._date_sortie

    def set_date_sortie(self, date_sortie: str):
        self._date_sortie = date_sortie

    def set_categorie(self, categorie: Categorie):
        self._categorie = categorie

    def get_categorie(self) -> Categorie:
        return self._categorie
    
    def set_realisateur(self, realisateur: "Realisateur"):
        self._realisateur = realisateur

    def get_realisateur(self) -> "Realisateur":
        return self._realisateur

    def description(self) -> str:
        if self._categorie:
            nom_cat = self._categorie.get_nom()
        else:
            nom_cat = "Inconnue"
            
        return f"Je suis un film de catégorie {nom_cat} et de durée {self._duree} sorti en {self._date_sortie}"